conans/test/integration/upload_test.py

```diff
@@ -0,0 +1,117 @@
+import unittest
+from conans.test.tools import TestServer, TestClient
+from conans.model.ref import ConanFileReference, PackageReference
+from conans.test.utils.cpp_test_files import cpp_hello_conan_files
+from nose.plugins.attrib import attr
+import os
+from conans.test.utils.test_files import uncompress_packaged_files
+
+
+class CompleteFlowTest(unittest.TestCase):
+
+    def setUp(self):
+        test_server = TestServer()
+        self.servers = {"default": test_server}
+        self.client = TestClient(servers=self.servers, users={"default": [("lasote", "mypass")]})
+
+    def reuse_uploaded_tgz_test(self):
+        '''Download packages from a remote, then copy to another channel
+        and reupload them. Because they have not changed, the tgz is not created
+        again'''
+
+        # UPLOAD A PACKAGE
+        conan_reference = ConanFileReference.loads("Hello0/0.1@lasote/stable")
+        files = cpp_hello_conan_files("Hello0", "0.1", need_patch=True, build=False)
+        files["another_export_file.lib"] = "to compress"
+        self.client.save(files)
+        self.client.run("export lasote/stable")
+        self.client.run("install %s --build missing" % str(conan_reference))
+        self.client.run("upload %s --all" % str(conan_reference))
+        self.assertIn("Compressing exported files", self.client.user_io.out)
+        self.assertIn("Compressing package", self.client.user_io.out)
+
+        # UPLOAD TO A DIFFERENT CHANNEL WITHOUT COMPRESS AGAIN
+        self.client.run("copy %s lasote/testing" % str(conan_reference))
+        self.client.run("upload Hello0/0.1@lasote/testing --all")
+        self.assertNotIn("Compressing exported files", self.client.user_io.out)
+        self.assertNotIn("Compressing package", self.client.user_io.out)
+
+        # THEN A NEW USER DOWNLOADS THE PACKAGES AND UPLOADS
+        # WITHOUT COMPRESSING AGAIN
+        other_client = TestClient(servers=self.servers, users={"default": [("lasote", "mypass")]})
+        other_client.run("install Hello0/0.1@lasote/stable --all")
+        self.client.run("upload Hello0/0.1@lasote/stable --all")
+        self.assertNotIn("Compressing exported files", self.client.user_io.out)
```

Comment:
I don't get it. I thought tgz were not being cached on download, so in this case, they should be compressed before the upload!